client/post-editor/term-token-field/index.jsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import cloneDeep from 'lodash/cloneDeep';
+import unescapeString from 'lodash/unescape';
+import { connect } from 'react-redux';
+import _debug from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPost } from 'state/posts/selectors';
+import { getTerms } from 'state/terms/selectors';
+import TokenField from 'components/token-field';
+import TermsConstants from 'lib/terms/constants';
+import PostActions from 'lib/posts/actions';
+import { recordStat, recordEvent } from 'lib/posts/stats';
+import QueryTerms from 'components/data/query-terms';
+
+const debug = _debug( 'calypso:post-editor:editor-terms' );
+
+class TermTokenField extends React.Component {
+	componentWillMount() {
+		this.boundOnTermsChange = this.onTermsChange.bind( this );
+	}
+
+	onTermsChange( selectedTerms ) {
+		debug( 'onTermsChange', selectedTerms, this );
+
+		let termStat, termEventLabel;
+		if ( selectedTerms.length > this.getPostTerms().length ) {
+			termStat = 'term_added';
+			termEventLabel = 'Added Term';
+		} else {
+			termStat = 'term_removed';
+			termEventLabel = 'Removed Term';
+		}
+		recordStat( termStat );
+		recordEvent( 'Changed Terms', termEventLabel );
+
+		const { postTerms, taxonomyName } = this.props;
+		const terms = cloneDeep( postTerms ) || {};
+		terms[ taxonomyName ] = selectedTerms;
+		// TODO: REDUX - remove flux actions when whole post-editor is reduxified
+		PostActions.edit( { terms } );
+	}
+
+	getPostTerms() {
+		const { postTerms, taxonomyName } = this.props;
+
+		if ( ! postTerms || ! postTerms[ taxonomyName ] ) {
+			return [];
+		}
+
+		if ( Array.isArray( postTerms[ taxonomyName ] ) ) {
+			return postTerms[ taxonomyName ];
+		}
+
+		return Object.keys( postTerms[ taxonomyName ] );
+	}
+
+	render() {
+		var termNames = ( this.props.terms || [] ).map( function( term ) {
+			return term.name;
+		} );
+
+		return (
+			<label className="editor-drawer__label">
+				<span className="editor-drawer__label-text">
+					{ this.props.taxonomyLabel }
+				</span>
+				<QueryTerms
+					siteId={ this.props.siteId }
+					taxonomy={ this.props.taxonomyName }
```

Comment:
Looking through the code some more, it doesn't look like it.  Pushed 0532210 to address this.  I'm having trouble fully testing the fix due to /me/sites timing out but will try again tomorrow.